apps/system/js/bluetooth_transfer.js

```diff
@@ -43,18 +36,28 @@ var BluetoothTransfer = {
     );
 
     // Listen to 'bluetooth-opp-transfer-start' from bluetooth.js
-    window.addEventListener('bluetooth-opp-transfer-start',
-      this.onUpdateProgress.bind(this, 'start')
-    );
+    window.addEventListener('bluetooth-opp-transfer-start', this);
 
     navigator.mozSetMessageHandler('bluetooth-opp-update-progress',
-      this.onUpdateProgress.bind(this, 'progress')
+      this._onUpdateProgress.bind(this, 'progress')
     );
 
     // Listen to 'bluetooth-opp-transfer-complete' from bluetooth.js
-    window.addEventListener('bluetooth-opp-transfer-complete',
-      this.onTransferComplete.bind(this)
-    );
+    window.addEventListener('bluetooth-opp-transfer-complete', this);
+  },
+
+  handleEvent: function bt_handleEvent(evt) {
+    switch(evt.type) {
+      case 'iac-bluetoothTransfercomms':
+        this._onFilesSending();
+        break;
+      case 'bluetooth-opp-transfer-start':
+        this._onUpdateProgress('start');
+        break;
+      case 'bluetooth-opp-transfer-complete':
+        this._onTransferComplete();
```

Comment:
Same here. I still like to keep original coding style. Please need to modify them beside of naming  for private methods. Thanks.